apple/swift-docc
# [Overloads] Do not automatically curate overloaded symbols, in favor of their overload group

Bug/issue #, if applicable: rdar://117768214

## Summary

The current implementation of "overload group symbols" as of https://github.com/apple/swift-docc/pull/846 currently curates the overload group alongside the individual overloaded symbols, creating a cluttered topics listing. This PR updates that logic to perform the following instead:

- Individual overloaded symbols are now hidden from automatic curation.
- The topic graph now tracks nodes' overload groups, and this information is used by the `SymbolGraphRelationshipsBuilder` to add additional edges between an overload group and individual overloaded symbols.
- The additional topic graph information is also queried by the externally-linkable element summary (for the `linkable-entities.json` digest file) to add an additional task group for an overload group's children, to allow tooling that consumes rendered archives to track overloaded symbols as part of their overload group.

## Dependencies

None

## Testing

Steps:
1. With a recent `main` branch Swift-DocC-Render built in `$DOCC_HTML_DIR`, run `swift run docc preview --enable-experimental-overloaded-symbol-presentation Tests/SwiftDocCTests/Test\ Bundles/OverloadedSymbols.docc`
2. Navigate to the `OverloadedEnum` symbol.
3. Ensure that there is only one `firstTestMemberName(_:)` instance method curated in the page.
4. Navigate to this `firstTestMemberName(_:)` method.
5. Ensure that there are no topic groups on this page.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
